src/main/java/org/traccar/helper/Checksum.java

```diff
@@ -200,4 +200,23 @@ public static long luhn(long imei) {
         return (10 - (checksum % 10)) % 10;
     }
 
+    public static int udp(ByteBuffer data) {
+        int sum = 0;
+        int len = data.capacity();
```

Comment:
Again, don't use capacity.

Reply:
ByteBuffer does not appear to have a length property/method. I have rewritten the for loop to while ByteBuffer.hasRemaining() however not sure what difference it makes.

Reply:
See comment above.